### PR TITLE
refactor(search.c): clint

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: src/nvim/search.c [hunk 1]
```diff
@@ -1765,9 +1763,8 @@ pos_T *findmatchlimit(oparg_T *oap, int initc, int flags, int64_t maxtravel)
               || STRNCMP(ptr, "el", 2) == 0) {
             hash_dir = 1;
           }
-        }
-        // Are we on a comment?
-        else if (linep[pos.col] == '/') {
+        } else if (linep[pos.col] == '/') {
+          // We are in a comment
```
Comment (zeertzjq) [inline src/nvim/search.c:1767]: Only a `/` does not mean a comment. Only `/*` and `*/` mean a comment.

File: src/nvim/search.c [hunk 2]
```diff
@@ -3767,7 +3764,7 @@ void find_pattern_in_path(char_u *ptr, Direction dir, size_t len, bool whole, bo
 
         if (compl_status_adding() && i == ins_compl_len()) {
           // IOSIZE > compl_length, so the STRNCPY works
-          STRNCPY(IObuff, aux, i);
```
Comment (zeertzjq) [inline src/nvim/search.c:3770]: Better add `NOLINT`

File: src/nvim/search.c [hunk 3]
```diff
@@ -3805,7 +3802,7 @@ void find_pattern_in_path(char_u *ptr, Direction dir, size_t len, bool whole, bo
             if (p - aux >= IOSIZE - i) {
               p = aux + IOSIZE - i - 1;
             }
-            STRNCPY(IObuff + i, aux, p - aux);
```
Comment (zeertzjq) [inline src/nvim/search.c:3808]: Better add `NOLINT`

File: src/nvim/search.c (hunk 2 above)
Comment (lewis6991) [reply src/nvim/search.c:3770]: How come? Do we want `STRNCPY` here?

Comment (zeertzjq) [reply src/nvim/search.c:3770]: It is from Vim isn't it?

Comment (lewis6991) [reply src/nvim/search.c:3770]: Yes, why does that matter?

Comment (zeertzjq) [reply src/nvim/search.c:3770]: Look at #6008. It seems that this has already been discussed.

Comment (justinmk) [reply src/nvim/search.c:3770]: > several call sites are depending on the zero-padding behavior of strncpy.

I'd leave the remaining strncpy calls alone.

File: src/nvim/search.c [hunk 4]
```diff
@@ -1765,9 +1763,8 @@ pos_T *findmatchlimit(oparg_T *oap, int initc, int flags, int64_t maxtravel)
               || STRNCMP(ptr, "el", 2) == 0) {
             hash_dir = 1;
           }
-        }
-        // Are we on a comment?
-        else if (linep[pos.col] == '/') {
+        } else if (linep[pos.col] == '/') {
+          // Are we on a comment?
```
Comment (justinmk) [inline src/nvim/search.c:1767]:
```suggestion
        } else if (linep[pos.col] == '/') {  // Are we on a comment?
```

File: src/nvim/search.c [hunk 5]
```diff
@@ -1331,9 +1330,8 @@ int do_search(oparg_T *oap, int dirc, int search_delim, char_u *pat, long count,
               break;
             }
           }
-        }
-        // to the left, check for start of file
-        else {
+        } else {
+          // to the left, check for start of file
```
Comment (justinmk) [inline src/nvim/search.c:1334]:
```suggestion
        } else {  // to the left, check for start of file
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**